dev-docs/RFCs/vNext/json-converter-keyword-handling-rfc.md

```diff
@@ -0,0 +1,27 @@
+# RFC: Syntactic sugar for @deck.gl/json
+
+* **Authors**: Andrew Duberstein (@ajduberstein)
+* **Date**: October 10, 2019
+* **Status**: Draft
+
+
+## Overview
+
+This is a proposal for adding syntactic sugar to the @deck.gl/json expression parser which will let the user distinguish
+between the keyword `type` and a string named `type`.
+
+
+## Motivation
+
+The keyword `type` occurs as a field name in many data sets, and is also of course a Javascript keyword.
+Users may want `type` to be interpreted as one or the other but currently can't specify which.
+
+The GeoJSON standard includes type in the top-level JSON, so most GeoJSON data sets break when ingested by the deck.gl/json API.
+
+## Proposal
+
+Suggestions from @ibgreen–
+
+Use `__type` to indicate the Javascript `type` keyword.
```

Comment:
Add a section on "Prior Art" or similar. Suggested text

### Prior Art

Have not found good similar examples. Maybe Vega JSON syntax has some constructs?

There are a few JSON-related formats that we may want to support in the future, so let's make sure we don't conflict with those.
- JSON pointer syntax - seems to starts with `/`
- JSON Paths, include `*` for wildcards, `@` for current object and 

Based on this, it looks like `_`, `__` (underscores) and `@@` would not conflict.

Underscores are somewhat more frequently used in JavaScript, though double underscores are very rare. Also, I have never encountered a JS variable starting with `@@`.

Maybe `@@` and `@@=` would stand out the most in the JSON text, flagging that these have special semantics.

Reply:
We use `__` for strictly internal fields [example 1](https://github.com/uber/deck.gl/blob/master/modules/core/src/lib/composite-layer.js#L92) [example 2](https://github.com/uber/deck.gl/blob/master/modules/core/src/lifecycle/component-state.js#L213).

Mapbox uses `$` to prefix special keywords, e.g. https://docs.mapbox.com/mapbox-gl-js/style-spec/#combining-filters

Reply:
Great input. 
- `__`: If we are already using `__` internally that would be a good reason to avoid it.
- `$`: I also feel that `$` is somewhat frequently used by other systems. We could certainly use that as an argument to align with that convention, but I my off-the-cuff sense is instead that we stay away from it, to minimize our risk of clashing with any other JSON format - in future APIs we may end up accepting such JSON snippets as props in order to forward to other libraries.